apps/src/applab/AppLabView.jsx

```diff
@@ -173,8 +164,16 @@ var AppLabView = React.createClass({
     return (
       <ConnectedStudioAppWrapper>
         <div id="visualizationColumn">
-          {playSpaceHeader}
-          <ProtectedStatefulDiv contentFunction={this.props.generateVisualizationColumnHtml} />
+          {!this.props.isReadOnlyWorkspace && <PlaySpaceHeader
+              isEditingProject={this.props.isEditingProject}
+              screenIds={this.props.screenIds}
+              onViewDataButton={this.props.onViewDataButton}
+              onScreenCreate={this.props.onScreenCreate} />
+          }
+          <ApplabVisualizationColumn
+              isProjectLevel={this.props.isProjectLevel}
+              isSubmittable={this.props.isSubmittable}
+              isSubmitted={this.props.isSubmitted}/>
```

Comment:
I think there's an interesting question about what should/should not be connected in this component.

As it exists right now, ApplabVisualizationColumn feels like it makes sense unconnected - but that might change if we move more logic inside of it. It also seems like PlaySpaceheader might better belong inside of ApplabVisualizationColumn.

TopInstructions has enough meat it in that it probably should be connected.

ConnectedStudioAppWrapper feels a little too simple to me to deserve it's own connection, and I would have expected us to just directly pass it the three props it needed.

The guidance in the redux docs is that connect should be used in container components, and not in presentational components. However, I'm not sure it's always clear which all of our components are - especially while we're still sort of in this mixed React/non-React mode.